### PR TITLE
feat(datastore) add OpenCensus spans to all Datastore clients

### PR DESCRIPTION
**Pull request by AlisskaPie**
1. `reserveIds`:
There wasn't any wrapping function for this operation, only generated in code, so I added one in client.go 
P.S. I guess some higher-level functions and tests for `reserveIds` for the following files could be added in another issue:
- integration_test.go
- key.go
- example_test.go 
- datastore_ test.go
2. `readWriteTransaction`
I couldn't find any function neither in the [documentation](https://cloud.google.com/datastore/docs/reference/data/rest), nor in code but I suspect we could add span for [read-write setting](https://cloud.google.com/datastore/docs/reference/data/rpc/google.datastore.v1#transactionoptions) in this [newTransaction](https://github.com/googleapis/google-cloud-go/blob/ceffc54ebb6159d1dff788238476c56135e104e9/datastore/transaction.go#L113) like it was done in [Java Pull Request. ](https://github.com/googleapis/java-datastore/pull/130/files)

### CODE REVIEW

File: datastore/transaction.go [hunk 1]
```diff
@@ -110,7 +110,10 @@ func (c *Client) NewTransaction(ctx context.Context, opts ...TransactionOption)
 	return c.newTransaction(ctx, newTransactionSettings(opts))
 }
 
-func (c *Client) newTransaction(ctx context.Context, s *transactionSettings) (*Transaction, error) {
+func (c *Client) newTransaction(ctx context.Context, s *transactionSettings) (_ *Transaction, err error) {
+	ctx = trace.StartSpan(ctx, "cloud.google.com/go/datastore.ReadWriteTransaction")
+	defer func() { trace.EndSpan(ctx, err) }()
+
```
Comment (IlyaFaer) [inline datastore/transaction.go]: `readWrite` is a transaciton type. On 118 line there is a condition `readOnly`. If transaction is not `readOnly` than it's `readWrite`, so it's the right place, the span lines just should be moved under the `if` statement

**Comment by IlyaFaer** [comment]
> There wasn't any wrapping function for this operation, only generated in code, so I added one in client.go
> P.S. I guess some higher-level functions and tests for reserveIds for the following files could be added in another issue:

That's the comment for a public PR. I think, it's okay to add a wrapper, but who knows

**Review by IlyaFaer — APPROVED** [review]

File: datastore/transaction.go (hunk 1 above)
Comment (AlisskaPie) [reply datastore/transaction.go]: Done

**Review by AlisskaPie — COMMENTED** [review]
Thank you

**Comment by AlisskaPie** [comment]
> > There wasn't any wrapping function for this operation, only generated in code, so I added one in client.go
> > P.S. I guess some higher-level functions and tests for reserveIds for the following files could be added in another issue:
> 
> That's the comment for a public PR. I think, it's okay to add a wrapper, but who knows

Yes, I wanted to send this whole comment actually with description

**Pull request closed without merging**